apps/minifront/src/components/swap/unclaimed-swaps.tsx

```diff
@@ -7,15 +7,23 @@ import { getSwapRecordCommitment } from '@penumbra-zone/getters/swap-record';
 import { uint8ArrayToBase64 } from '@penumbra-zone/types/base64';
 import { GradientHeader } from '@repo/ui/components/ui/gradient-header';
 import { useStoreShallow } from '../../utils/use-store-shallow';
+import { useState } from 'react';
+import { SwapRecord } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
 
 const unclaimedSwapsSelector = (state: AllSlices) => ({
   claimSwap: state.unclaimedSwaps.claimSwap,
-  isInProgress: state.unclaimedSwaps.isInProgress,
 });
 
 export const UnclaimedSwaps = () => {
   const unclaimedSwaps = useUnclaimedSwaps();
-  const { claimSwap, isInProgress } = useStoreShallow(unclaimedSwapsSelector);
+  const { claimSwap } = useStoreShallow(unclaimedSwapsSelector);
+  const [claim, setClaim] = useState<string[]>([]);
+
+  // Internal state management for tracking the IDs of the swaps that are currently being claimed.
+  const handleClaim = async (id: string, swap: SwapRecord) => {
+    setClaim(previous => [...previous, id]);
+    await claimSwap(id, swap);
```

Comment:
question: would it be useful to remove the `id` from `claim` in case the error happens in `claimSwap` function?

Reply:
this is a good point, I modified it that if the `claimSwap` throws an error, we remove the id from the claim array, allowing the button to be clicked again for claiming the swap.